parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -240,9 +241,32 @@ extractKubeBinaries() {
     rm -f "$K8S_DOWNLOADS_DIR/${K8S_TGZ_TMP}"
 }
 
-installKubeletKubectlAndKubeProxy() {
+extractPrivateKubeBinaries() {
+    K8S_VERSION=$1
+    KUBE_BINARY_URL=$2
+
+    K8S_TGZ_TMP=${KUBE_BINARY_URL##*/}
+    CACHED_PKG="${K8S_CACHE_DIR}/${K8S_TGZ_TMP}"
+
+    if [[ -f "${CACHED_PKG}" ]]; then
+        echo "cached package ${CACHED_PKG} is found, will use that"
+    else
+        echo "cached package ${CACHED_PKG} not found"
+        return 1
+    fi
+
+    retrycmd_get_tarball 120 5 "${CACHED_PKG}" ${KUBE_BINARY_URL} || exit $ERR_K8S_DOWNLOAD_TIMEOUT
```

Comment:
should we have a new exit code for failures when downloading private kube binaries?

Reply:
I think it is no different than any other package and for internal only usage once in a while - should be obvious when failed? Do we track it codes/failures via any other automation?

Reply:
we track CSE failures by exit code in our node SIG dashboard - I think it might be useful to be able to differentiate failures on normal and private cases

Reply:
updated with ERR_PRIVATE_K8S_PKG_ERR=203